metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -197,20 +197,15 @@ func getMasterName(http *helper.HTTP, uri string) (string, error) {
 }
 
 // GetInfo returns the data for the Elasticsearch / endpoint.
-func GetInfo(http *helper.HTTP, uri string) (*Info, error) {
-
+func GetInfo(http *helper.HTTP, uri string) (Info, error) {
+	info := Info{}
 	content, err := fetchPath(http, uri, "/", "")
 	if err != nil {
-		return nil, err
+		return info, err
 	}
 
-	info := &Info{}
 	err = json.Unmarshal(content, &info)
-	if err != nil {
-		return nil, err
-	}
-
-	return info, nil
+	return info, err
 }
```

Comment:
Same thing as in the other one:


```suggestion
func GetInfo(m *MetricSet) (Info, error) {
	content, err := fetchPath(m.HTTP, "/")
	if err != nil {
		return Info{}, err
	}

	info := Info{}
	if err:= json.Unmarshal(content, &info); err != nil {
		return Info{}, err
	}

	return info, err
}
```